index.js

```diff
@@ -10,6 +10,20 @@ function isAgent(v) {
   return v && typeof v.addRequest === 'function';
 }
 
+function urlInstanceToObject(url) {
+  var urlObject = {};
+  // If the opts are genearted with the WHATWG API
+  if (url && Object.getOwnPropertySymbols && Object.getOwnPropertySymbols(url)[0]) {
```

Comment:
Some more comments or pointers explaining this magic would be very useful for the future reader. Even now I don't fully understand what is happening :D

Reply:
I'll add comments for sure! First here, since it's easier:

This line is about checking whether the `url` is a WHATWG url or not. I'm doing this by first checking that the url exists (is not falsy), then checking that Object.getOwnPropertySymbols exists (since this feature appeared relatively recently, compared the versions of Node that are expected to work on CI), and finally by combining this Object method with the given URL to confirm that it indeed has symbols.

I currently don't know a better way of doing this! I pushed something fast that seemed to work. I'm interested in learning how to make this better while updating this as fast as I can! 🌻

What do you think of this approach? Any suggestions?